### ISSUE TITLE
Inconsistent dimension bug in TriAttention

### ISSUE THREAD

**Issue by zhongpeixiang** [aioz-ai/ICCV19_VQA-CTI#5]
When computing the mask at https://github.com/aioz-ai/ICCV19_VQA-CTI/blob/e1f93d39b8e2bb63a1b27027620803d713d23eaa/src/attention.py#L55

``v`` has 3 dimensions, ``logits`` has 4 dimensions, thus, computing ``mask`` results in an error.

BTW, what does ``mask`` mean? why use ``(0 == v.abs().sum(2))`` as the mask?

Thanks
